src/device/mod.rs

```diff
@@ -689,6 +689,84 @@ impl RequestedState {
             })
             .collect()
     }
+
+    // Same as collect_dns_records() fn but for peers with defined nicknames.
+    pub fn collect_dns_nickname_records(&self) -> Records {
```

Comment:
I forgot to add ".nord" :sweat:

Reply:
This could be final version
```rust
        let to_record = |p: &PeerBase| {
            p.nickname.as_ref().cloned().and_then(|nick| {
                p.ip_addresses
                    .as_ref()
                    .cloned()
                    .and_then(|ips| Some((nick, ips)))
            })
        };

        self.meshnet_config
            .iter()
            .flat_map(|cfg| {
                to_record(&cfg.this).into_iter().chain(
                    cfg.peers
                        .iter()
                        .flat_map(|peers| peers.iter())
                        .filter_map(|p| to_record(p)),
                )
            })
            .filter(|(nick, _)| self.validate_record(nick))
            .fold(HashMap::new(), |mut map, (mut key, value)| {
                key += ".nord";
                if map.contains_key(&key) {
                    telio_log_warn!("Nickname is already assigned: {key:?}. Ignore");
                } else {
                    map.insert(key, value);
                }
                map
            })
```

Reply:
nice! will use it